bootstrap/kubeadm/api/v1alpha4/kubeadmbootstrapconfig_types.go

```diff
@@ -95,6 +98,34 @@ type KubeadmConfigSpec struct {
 	// For more information, refer to https://github.com/kubernetes-sigs/cluster-api/pull/2763#discussion_r397306055.
 	// +optional
 	UseExperimentalRetryJoin bool `json:"useExperimentalRetryJoin,omitempty"`
```

Comment:
I think so. We've been running tests in CAPA without it for a while now without ill effects. CC @yastij and @gab-satchi who can speak to vSphere.

Reply:
Ultimately depends on our support matrix K8s versions in v1alpha4.

Reply:
yeah, IIRC we had fixes in k/k related to etcd but not sure they were backported to older version @fabriziopandini 

options we have today:
- graduate this and keep default to false
- remove the field and the retries
- remove the field and retry on failure

I still think that this has value, especially in cases where you have a slow connectivity

Reply:
from the kubeadm side things were backported, but if you have a specific issue in mind I can double check.
WRT to the options, given that this feature was added following after a round of test on slow infrastructures, It will be really useful repeat the same set of test so we can take an informed decision

Reply:
@vincepri can we mark this as resolved?